doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,889 @@
+    {
+        /// The default experience for the installer. Installer may show some UI.
+        Default,
+        /// Runs the installer in silent mode. This suppresses the installer's UI to the extent 
+        /// possible (installer may still show some required UI).
+        Silent,
+        /// Runs the installer in interactive mode.
+        Interactive,
+    };
+
+    /// Options when installing a package.
+    /// Intended to allow full compatibility with the "winget install" command line interface.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass InstallOptions
+    {
+        InstallOptions();
+
+        /// DESIGN NOTE: --Exact from the winget install command line is implied here by virtue of the find 
+        /// then install object semantics.
+        /// Likewise --Query from the command line is not implemented here since the query is required up 
+        /// front in order to get the CatalogPackage object.
+        /// DESIGN NOTE: Using PackageVersionInfo rather than CatalogPackage felt like it would be too 
+        /// specific here. CatalogPackage and PackageVersionId (or if unspecified then the latest one) may be nicer
+        /// if we end up having versions that have different applicability or other more complex logic about 
+        /// which version is preferred.
+        CatalogPackage CatalogPackage;
+        /// Optionally specifies the version from the package to install. If unspecified the version matching 
+        /// CatalogPackage.GetLatestVersion() is used.
+        PackageVersionId PackageVersionId;
+
+        /// Specifies alternate location to install package (if supported).
+        String PreferredInstallLocation;
+        /// User or Machine. 
+        AppInstallScope AppInstallScope;
+        /// Silent, Interactive, or Default
+        AppInstallMode AppInstallMode;
+        /// Directs the logging to a log file. Callers must provide a path to a file that the Windows Package 
+        /// Manager package has write access to.
+        String LogOutputPath;
+        /// Continues the install even if the hash in the catalog does not match the linked installer.
+        Boolean AllowHashMismatch;
+        /// A string that will be passed to the installer. 
+        /// IMPLEMENTATION NOTE: maps to "--override" in the winget cmd line
+        String ReplacementInstallerArguments;
+
+        /// Used by a caller to correlate the install with a caller's data.
+        /// IMPLEMENTATION NOTE: This area is in flux but right now it's expected that this would allow a caller to 
+        /// use JSON to encode multiple arguments.
+        String CorrelationData;
+        /// A string that will be passed to the source server if using a REST source
+        String AdditionalAppCatalogArguments;
+    }
+
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass AppInstaller
+    {
+        AppInstaller();

```

Comment:
If we make this a proper Windows Runtime class, then this can become a static class.

Reply:
Can do this on the client side if we do a client wrapper.

Reply:
DISCUSSION: See earlier comments about CoCreate vs RoActivate here